Update readme (#2)
DevExpress-Examples/xaf-how-to-implement-a-custom-attribute-to-customize-the-application-model: Readme.md

<!-- default badges list -->
[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E1679)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# XAF - How to implement a custom attribute to customize the Application Model

This example demonstrates how to create a custom attribute and remove a property from the [Application Model](https://docs.devexpress.com/eXpressAppFramework/112579/ui-construction/application-model-ui-settings-storage)'s `DetailView.Items` and `ListView.Columns` nodes.
In this case, the properties marked with this attribute are removed from views.

## Implementation Details

> To implement this functionality, the `IModelMember` interface is extended with the `IRemovedFromViewInfo` interface that contains a single property - `IsRemovedFromViewInfo`. The value of this property is calculated based on the value of the custom attribute we created (`RemoveFromViewInfoAttribute`) and applied to the persistent class. This is done via the domain logic (`RemovedFromViewInfoLogic`). The `IRemovedFromViewInfo.IsRemovedFromViewInfo` property is used by the model node generator updater (`ViewsNodesGeneratorUpdater`) to remove certain view items and columns from detail and list views. 

Reviewer: ```suggestion
To implement this functionality, the `IModelMember` interface is extended with the `IRemovedFromViewInfo` interface that contains the `IsRemovedFromViewInfo` property. The value of this property is calculated based on the value of the custom attribute we created (`RemoveFromViewInfoAttribute`) and applied to the persistent class via the domain logic (`RemovedFromViewInfoLogic`). The `IRemovedFromViewInfo.IsRemovedFromViewInfo` property is used by the model node generator updater (`ViewsNodesGeneratorUpdater`) to remove certain view items and columns from detail and list views. 
```

> Thus, the example demonstrates three techniques: extension of the application model with additional properties, definition of default model values via domain logic, and modification of the default model via the model node generator updater.

Reviewer: ```suggestion
This example demonstrates three techniques: how to extend the application model with additional properties, define default model values via domain logic, and modify the default model via the model node generator updater.
```
